src/components/Card/Card.jsx

```diff
@@ -1,14 +1,20 @@
 import React from 'react';
+import { connect } from 'react-redux';
 import styles from './Card.styl';
 import PropTypes from 'prop-types';
+import { setMovieDetailsThunk } from 'store/actions';
 import CardImage from '../CardImage/CardImage';
 import CardFooter from '../CardFooter/CardFooter';
 import { Link } from 'react-router-dom';
 
-const Card = ({ img, title, subtitle, label, id, rating }) => {
+const Card = ({ img, title, subtitle, label, id, rating, setMovieDetailsThunk }) => {
+  const handleClick = () => {
+    setMovieDetailsThunk(id);
+  }
+
   return (
     <div className={styles.Card}>
-      <Link to={`/movie/${id}`}>
+      <Link onClick={handleClick} to={`/movie/${id}`}>
```

Comment:
can you guarantee that `onClick={handleClick}` and `to={`/movie/${id}`}` won't interrupt each other?